WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -384,17 +386,79 @@ private async Task CreateBlameRoundAsync(Round round, CancellationToken cancella
+			&& Rounds.Any(x => // If there isn't any input reg round, then let the software create a normal round.
+				x.Phase == Phase.InputRegistration
+				&& x is not BlameRound
+				&& !x.IsInputRegistrationEnded(Config.MaxInputCountByRound)))
+		{
+			feeRate = (await Rpc.EstimateSmartFeeAsync((int)Config.ConfirmationTarget, EstimateSmartFeeMode.Conservative, simulateIfRegTest: true, cancellationToken).ConfigureAwait(false)).FeeRate;
+			RoundParameters parameters = RoundParameterFactory.CreateRoundParameter(feeRate, MaxSuggestedAmountProvider.MaxSuggestedAmount);
+
+			TryCreateFirstOrderedRound(parameters);
+			LastFirstOrderedRoundCreated = DateTimeOffset.UtcNow;
+		}
+
+		// Add more rounds if not enough.
 		var registrableRoundCount = Rounds.Count(x => x is not BlameRound && x.Phase == Phase.InputRegistration && x.InputRegistrationTimeFrame.Remaining > TimeSpan.FromMinutes(1));
-		int roundsToCreate = Config.ParallelRounds - registrableRoundCount;
+		int roundsToCreate = Config.RoundParallelization - registrableRoundCount;
 		for (int i = 0; i < roundsToCreate; i++)
 		{
-			var feeRate = (await Rpc.EstimateSmartFeeAsync((int)Config.ConfirmationTarget, EstimateSmartFeeMode.Conservative, simulateIfRegTest: true, cancellationToken).ConfigureAwait(false)).FeeRate;
+			feeRate ??= (await Rpc.EstimateSmartFeeAsync((int)Config.ConfirmationTarget, EstimateSmartFeeMode.Conservative, simulateIfRegTest: true, cancellationToken).ConfigureAwait(false)).FeeRate;
+			RoundParameters parameters = RoundParameterFactory.CreateRoundParameter(feeRate, MaxSuggestedAmountProvider.MaxSuggestedAmount);
 
-			RoundParameters parameters =
-				RoundParameterFactory.CreateRoundParameter(feeRate, MaxSuggestedAmountProvider.MaxSuggestedAmount);
-			Round r = new(parameters, SecureRandom.Instance);
+			var r = new Round(parameters, SecureRandom.Instance);
+			Rounds.Add(r);
+			r.LogInfo($"Created round with params: {nameof(r.Parameters.MaxSuggestedAmount)}:'{r.Parameters.MaxSuggestedAmount}' BTC.");
+		}
+	}
+
+	private void TryCreateFirstOrderedRound(RoundParameters parameters)
+	{
+		// Huge HACK to keep it compatible with WW2.0.0 client version, which's
+		// round preference is based on the ordering of ToImmutableDictionary.
+		// Add round until ToImmutableDictionary orders it to be the first round
+		// so old clients will prefer that one.
+		IOrderedEnumerable<Round>? orderedRounds;
+		Round r;
+		var before = DateTimeOffset.UtcNow;
+		var times = 0;
+		var maxCycleTimes = 300;
+		do
+		{
+			var roundsCopy = Rounds.ToList();
+			r = new Round(parameters, SecureRandom.Instance);
+			roundsCopy.Add(r);
+			orderedRounds = roundsCopy
+				.Where(x => x.Phase == Phase.InputRegistration && x is not BlameRound && !x.IsInputRegistrationEnded(Config.MaxInputCountByRound))
+				.OrderBy(x => x.InputCount);
+			times++;
+		}
+		while (times <= maxCycleTimes && orderedRounds.ToImmutableDictionary(x => x.Id, x => x).First().Key != r.Id);
```

Comment:
I don't quite understand this.
Newly created Round `r`'s input count will be 0.
Then we order all rounds by input count, so `r` should be the first with 0.
Then we check `r.id != r.id` but it will be false.
Then we redo everything again untill we reach 300 tries?
Am I missing something?

Reply:
`ToImmutableDictionary` changes the order. This runs on the client side, so whatever order the client gets the rounds, they'll `ToImmutableDictionary` and that'll change the order for them.

> Then we order all rounds by input count, so r should be the first with 0.

This is probably unnecessary, we do it just to be consistent with the getstatus request's ordering.

Reply:
Mining a round! How it didn't come to my mind!? Very good.
The mambo yambo with the list is all unnecessary but anyway, congrats.

Reply:
Doesn't do the job.  On the testnet we were coming different times when we tested, but on the mainnet users are selecting the round the same time :( We're gonna try make this work tomorrow: https://github.com/zkSNACKs/WalletWasabi/pull/8502